cartographer/io/proto_stream_test.cc

```diff
@@ -32,9 +33,13 @@ namespace {
 class ProtoStreamTest : public ::testing::Test {
  protected:
   void SetUp() override {
-    const std::string tmpdir = P_tmpdir;
-    test_directory_ = tmpdir + "/proto_stream_test_XXXXXX";
-    ASSERT_NE(mkdtemp(&test_directory_[0]), nullptr) << strerror(errno);
+    auto testdir = boost::filesystem::temp_directory_path();
```

Comment:
The plan was to get rid of boost completely. Please, do not use boost::filesystem.

Reply:
`boost::filesystem` was merged into C++17 (`std::filesystem`). Here I'm using it just to get the temp directory, and I think we need to use it in other places as well. Here's why: it provides a cross-platform `path` class which can handle UTF paths both on Windows and Linux (it needs to be a wide UTF16 string on Windows due to the way Windows handles Unicode). 

So if you have a .lua tuning named `チューニング.lua`, you're not going to open it without `boost::filesystem` :)

Reply:
I am sure, we can live without names like "チューニング.lua" until we enable c++17.

Reply:
Keep in mind it's enough just to have a UTF username in the path to trigger this. For example, `C:\Users\Александrр`, and you can't load/save anything from Desktop or My Documents, which I'd say is quite broken. I have a 10-liner fixup for this ready, just if you're willing to link against boost::filesystem.

Reply:
Does this lead to an error in the Windows CI pipeline or only if you run the test on a user PC with UTF signs in the paths?

Reply:
Actually, the UTF chars in path thing is not related to this PR, it was just my line of reasoning why we should adopt `boost::filesystem`.

In this PR, we just need a temporary directory path. The test was previously using `P_tmpdir` for this, which is not available on Windows since it's a gnu libc-specific global variable. `boost::filesystem` provides `temp_directory_path()` which is a suitable cross-platform replacement.

Reply:
A patch which would fix loading .lua files with unicode paths would look like this:
https://pastebin.com/bgJv1jNz

Reply:
I am not willing to link against boost for many reasons. One of them is that our internal version of Cartographer will be broken.

Reply:
In that case, what should we do with `mkdtemp` and `P_tmpdir` here?

Reply:
Is there something in Abseil we could use?

Reply:
Not yet. the file lib will be there only next year.

Reply:
> In that case, what should we do with mkdtemp and P_tmpdir here? These are linux/glibc specific.

Damn. Okay, this is still relevant.

Reply:
Maybe we can just create the test files in the current working directory and avoid this mess with cross-platform temporary directory paths.